flytekit/core/interface.py

```diff
@@ -381,10 +382,12 @@ def transform_function_to_interface(fn: typing.Callable, docstring: Optional[Doc
     return_annotation = type_hints.get("return", None)
 
     ctx = FlyteContextManager.current_context()
-    # Only check if the task/workflow has a return statement at compile time locally.
     if (
         ctx.execution_state
+        # Only check if the task/workflow has a return statement at compile time locally.
         and ctx.execution_state.mode is None
+        # inspec module does not work correctly with Python <3.10. https://github.com/flyteorg/flyte/issues/5608
+        and sys.version_info >= (3, 10)
```

Comment:
Should we compare the micro level here too?

Reply:
Thank you for the quick reaction @pingsutw btw 🙇

Reply:
nice, good call. updated it.